src/store-api/src/metadata.rs

```diff
@@ -741,92 +718,114 @@ impl RegionMetadataBuilder {
         Ok(())
     }
 
-    fn change_column_inverted_index_options(
-        &mut self,
-        column_name: String,
-        value: bool,
-    ) -> Result<()> {
+    fn set_indexes(&mut self, options: Vec<SetIndexOption>) -> Result<()> {
+        let mut index_options = options
+            .into_iter()
+            .map(|option| (option.column_name().clone(), option))
+            .collect::<HashMap<_, _>>();
+
         for column_meta in self.column_metadatas.iter_mut() {
-            if column_meta.column_schema.name == column_name {
-                column_meta.column_schema.set_inverted_index(value)
+            if let Some(option) = index_options.remove(&column_meta.column_schema.name) {
+                Self::set_index(column_meta, option)?;
             }
         }
```

Comment:
There will be a very large overhead when there are many columns. How about:
- don't collect into a hashmap
- iterate over the request, not the column metadatas
- also applies to `unset_indexes`